### DISCUSSION TITLE
RuntimeError: No application was set for this JobQueue.

### DISCUSSION THREAD

**Discussion by ldnora** [python-telegram-bot/python-telegram-bot#3725] [Q&A] [answered]
Greetings!

Firstly, i need to thank the work that the team are developing in the time. It's great value.

I'm work to actual a bot that verify the server and the network at a university in Brasil. For this, in terms of code, i use the class JobQueue.run_daily, together with the appropriate parameters defined in the documentation. But, by running the code, i receive this error: 

```
bot-telegram    | Traceback (most recent call last):
bot-telegram    |   File "/application/main.py", line 39, in <module>
bot-telegram    |     bot_admin.ExecAutoCommands()
bot-telegram    |   File "/application/src/Bot.py", line 567, in ExecAutoCommands
bot-telegram    |     self.jobs.run_daily(self.TestServers, horario_execucao)
bot-telegram    |   File "/usr/local/lib/python3.10/site-packages/telegram/ext/_jobqueue.py", line 528, in run_daily
bot-telegram    |     args=(self.application,),
bot-telegram    |   File "/usr/local/lib/python3.10/site-packages/telegram/ext/_jobqueue.py", line 155, in application
bot-telegram    |     raise RuntimeError("No application was set for this JobQueue.")
bot-telegram    | RuntimeError: No application was set for this JobQueue.
```
Note: bot_admin.ExecAutoCommands() contains the JobQueue.run_daily.

To underestend the class, i discovered that a variable (self._application) recive None and it isn't modified to the long of the code.

_jobqueue.py:
```python
[...]
def __init__(self) -> None:
    [...]
    self._application: "Optional[weakref.ReferenceType[Application]]" = None
    [...]

def application(self) -> "Application":
    """The application this JobQueue is associated with."""
    if self._application is None:`
        raise RuntimeError("No application was set for this JobQueue.")
        [...]`

def run_daily
    [...]
         j = self.scheduler.add_job(
        [...]
        args=(self.application,),
        [...]
```
		
By investigatin this such error in the web, i not found nothing in this regasd. I think it's current by codumentation, even tough i have reviewd. I think i'm using the class in the wrong way. 	
Use Ubunto v22.4 LTS, python-telegram-bot v20.2 and python v3.10.

Thank you in advance for your help! 
Great day!

**Comment by Bibo-Joshi** [comment, marked as answer]
Please see [here](https://docs.python-telegram-bot.org/en/stable/#optional-dependencies), [here](https://github.com/python-telegram-bot/python-telegram-bot/wiki/Extensions---JobQueue#usage) and the "Important" note [here](https://docs.python-telegram-bot.org/en/stable/telegram.ext.jobqueue.html).
